### DISCUSSION TITLE
Disable ID response check

### DISCUSSION THREAD

**Discussion by davekeeshan** [alexforencich/cocotbext-axi#66] [General]
Is it possible to disable the ID response check?

`assert self.active_id[rid] > 0, "unexpected burst ID"`

I am currently trying to simulate against some xilinx IP (I ultimately plan to rewrite this IP, but that is down the road) but the id feedback in most xilinx IP is just not implemented, so it crashes all my sims.  If it were possible to disable the check or mark the it down to warning, not exception that would be very helpful.

I have a hack right now, but it is awful.

**Comment by alexforencich** [comment]
Well, not handling the ID signal correctly is a serious AXI protocol violation, definitely an assertion-failure-level problem, a warning is not sufficient here.  Like, what should the simulation model do if it gets a response on the R/B channel with an ID value that it never used?  Drop it, likely causing a hang?  Assume it's in response to something that it did issue, likely resulting in mismatched read data?  There is no clean solution here, anything other than an assert failure would be a horrible hack.

What, exactly is the Xilinx IP doing in this case?  Returning everything in order, but with ID 0?  If there is no reordering, then the ID signal isn't really necessary.  In that case, you could simply force all of the operations to use AWID 0/ARID 0 by adding that as a parameter (e.g. `inst.read(addr, len, arid=0)` or `inst.write(data, awid=0)`).

**Comment by davekeeshan** [comment]
It is a hard one for me as I am using your code to test xilinx code, so I can't change what xilinx are doing.  Just looking at their basic uart with and axi interface, they take in the id, but do not pass it back.  In terms of failure, well if you stay inside the xilinx ecosystem and they are all performing the same way, then it all "works".  (There are a lot of cases where the xilinx IP is a bit generous with protocols) The issue gets further confused if you use their axi_crossbar, it may have a id width of 4, but have peripherals with an id width of 1.  

Ultimately I am working on a design that will start in a fpga and then need to migrate out to an asic, so xilinx IP is a dead end for me.

My personal goal is to actually write some open synthesisable/simulatable replacements, but I want to write the tests first using the IP as a guide, emitting the warnings is fine, stopping the simulation dead is where I hit issues.  As you said and I did, the hack i had to implement is horrible.

However, I am glad to say what you recommend is perfect, I did that, rolled out my hack and it all works so happy days.
